File: stdio.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#define true 1
#define false 0
int file_input=false;
int file_output=false;
int file_error=false;
void file_open(int i){
    if(i==0)file_input=true;
    if(i==1)file_output=true;
    if(i==2)file_error=true;

}
void init(){
    file_open(2);
    file_open(1);
    file_open(2);

}
void file_write(int f1,char *c){
    if(f1==1 && file_output)printf("stdout:>%s\n",c);
    if(f1==2 && file_output)printf("stderr:>%s\n",c);
}
void file_close(int i){
    if(i==0)file_input=false;
    if(i==1)file_output=false;
    if(i==2)file_error=false;


}
int main(){
   printf("\033c\033[43;30m\n");
   init();
   file_write(1,"hello world...");
   file_write(2,"error message....");
   file_close(1);
   file_write(1,"hello world...");
  
   return 0;
}

File: stdio.py

global file_input
global file_output
global file_error
file_input=False
file_output=False
file_error=False
def opens(handlers):
    global file_input
    global file_output
    global file_error
    if handlers == 0:
        file_input=True
    if handlers == 1:
        file_output=True
    if handlers == 2:
        file_error=True

def closes(handlers):
    global file_input
    global file_output
    global file_error
    if handlers == 0:
        file_input=False
    if handlers == 1:
        file_output=False
    if handlers == 2:
        file_error=False

def writes(handlers:int,s:str):
        global file_input
        global file_output
        global file_error
        if handlers == 1 and file_output:
            print("stdout:>"+s)
        if handlers == 2 and file_error:
           print("stderr:>"+s)


def clear():
    print("\033c\033[43;30m\n")
    opens(0)
    opens(1)
    opens(2)

clear()
writes(1,"hello world...")
writes(2,"error tests")
closes(2)
closes(1)
writes(1,"hello world...")
writes(2,"error tests")

